vainfo/vainfo.c

```diff
@@ -84,19 +91,421 @@ parse_args(const char *name, int argc, char **argv)
     }
 }
 
+static int show_config_attributes(VADisplay va_dpy, VAProfile profile, VAEntrypoint entrypoint)
+{
+  struct str_format
+  {
+    int format;
+    char *name;
+  };
+
+  VAStatus va_status;
+  int max_num_attributes, i, n;
+  VAConfigAttrib * attrib_list = NULL;
+  int ret_val = 0;
+
+  max_num_attributes = vaMaxNumConfigAttributes(va_dpy);
+  attrib_list = malloc (max_num_attributes * sizeof (VAConfigAttrib));
+  for (i = 0; i < max_num_attributes; i++) {
+      attrib_list[i].type = i;
+  }
+
+  va_status = vaGetConfigAttributes(va_dpy,
```

Comment:
Per https://github.com/intel/libva/blob/95eb8cf469367b532b391042fa0e77ca513ac94e/va/va.h#L1195 you just need to use static array and you will get VA_ATTRIB_NOT_SUPPORTED for not supported attributes:
```
VAConfigAttrib attrib_list[VAConfigAttribTypeMax];

vaGetConfigAttributes(..., attrib_list, sizeof(attrib_list)/sizeof(VAConfigAttribTypeMax));
```

Reply:
Right. Thanks